From 2a56ea6ea2eb79c280ed64ec0ca842c4bcb6ce71 Mon Sep 17 00:00:00 2001
From: Emergi Idan <idan.emergi@intel.com>
Date: Sun, 10 Nov 2019 15:44:59 +0200
Subject: [PATCH] PONRTSYS-6265: fix modifier recipes core commands APIs

previous fix didn't cover all recipes and commands
this commit fixes all recipes commands which make use of the pre L2
size as a legitimate layer2 offset which cause calculations
error in the modifier recipes, therfore, causeing  the robustness violation.

Signed-off-by: Emergi Idan <idan.emergi@intel.com>
---
 drivers/net/ppv4/pp-hal/modifier/modifier_core.c     | 10 +++++-----
 drivers/net/ppv4/pp-hal/modifier/modifier_internal.h | 12 +++++-------
 2 files changed, 10 insertions(+), 12 deletions(-)

diff --git a/drivers/net/ppv4/pp-hal/modifier/modifier_core.c b/drivers/net/ppv4/pp-hal/modifier/modifier_core.c
index 65b5c82812b5..f2b49b3dde02 100644
--- a/drivers/net/ppv4/pp-hal/modifier/modifier_core.c
+++ b/drivers/net/ppv4/pp-hal/modifier/modifier_core.c
@@ -393,7 +393,7 @@ s32 mod_bce_ip_nat_cpy(struct mod_recipe *rcp, s32 bce_off, s32 dst_layer)
 	/* SI1 = SI[si_mux1] (0) */
 	eu.si_mux1 = 0;
 	/* STW0 = L2 off (0) */
-	eu.stw_mux0 = EU_STW_SEL_L2_OFF;
+	eu.stw_mux0 = EU_STW_SEL_ZERO;
 	/* STW1 = SW[stw_mux1] */
 	eu.stw_mux1 = dst_layer;
 
@@ -463,7 +463,7 @@ s32 mod_reg_to_hdr_cpy(struct mod_recipe *rcp, s32 src, s32 layer, u8 off,
 	/* STW1 = layer offset from STW MUX */
 	eu.stw_mux1 = layer;
 	/* stw0 = L2 from STW MUX */
-	eu.stw_mux0 = EU_STW_SEL_L2_OFF;
+	eu.stw_mux0 = EU_STW_SEL_ZERO;
 	eu.jm_store = EU_STORE_SI1_SW1_NO_SW0;
 
 	/* size */
@@ -509,7 +509,7 @@ s32 mod_reg_to_new_hdr_cpy(struct mod_recipe *rcp, s32 src, s32 layer, u32 off,
 	/* SI1 = layer in new header */
 	eu.si_mux1    = layer;
 	/* stw1 = L2 from STW MUX */
-	eu.stw_mux1   = EU_STW_SEL_L2_OFF;
+	eu.stw_mux1   = EU_STW_SEL_ZERO;
 	eu.jm_store   = EU_STORE_SI1_SW1;
 
 	/* size */
@@ -649,7 +649,7 @@ s32 mod_alu_synch(enum mod_eu_type alu, struct mod_recipe *rcp, s32 layer,
 		eu.si_mux0 = EU_SI_SEL_ZERO;
 	}
 	eu.stw_mux1   = layer;/* SW1 = SW[stw_mux1] */
-	eu.stw_mux0   = EU_STW_SEL_L2_OFF;/* SW0 = SW[stw_mux0] */
+	eu.stw_mux0   = EU_STW_SEL_ZERO;/* SW0 = SW[stw_mux0] */
 	/* use load0 + SI0 + SW1 - SW0 */
 	eu.jm_load0   = EU_LOAD0_SI0_SW1_NO_SW0;
 	eu.jm_chnk_sz = EU_CHNK_SZ_RAW;
@@ -692,7 +692,7 @@ s32 mod_pkt_to_reg_cpy(struct mod_recipe *rcp, s32 dst, s32 layer, u8 off,
 		eu.si_mux0 = EU_SI_SEL_ZERO;
 	}
 	eu.stw_mux1 = layer;/* SW1 = SW[stw_mux1] */
-	eu.stw_mux0 = EU_STW_SEL_L2_OFF;/* SW0 = SW[stw_mux0] */
+	eu.stw_mux0 = EU_STW_SEL_ZERO;/* SW0 = SW[stw_mux0] */
 	/* use load0 + SI0 + SW1 - SW0 */
 	eu.jm_load0 = EU_LOAD0_SI0_SW1_NO_SW0;
 	eu.store = dst;
diff --git a/drivers/net/ppv4/pp-hal/modifier/modifier_internal.h b/drivers/net/ppv4/pp-hal/modifier/modifier_internal.h
index 49e0a5fda3cd..624b529d976f 100644
--- a/drivers/net/ppv4/pp-hal/modifier/modifier_internal.h
+++ b/drivers/net/ppv4/pp-hal/modifier/modifier_internal.h
@@ -298,8 +298,8 @@ enum mod_eu_stw_mux_sel {
 	EU_STW_SEL_L3_OFF1,
 	EU_STW_SEL_L3_OFF0,
 	EU_STW_SEL_PRSR_PKT_TYPE,
-	EU_STW_SEL_L2_OFF,
-	EU_STW_SEL_LAST    = EU_STW_SEL_L2_OFF,
+	EU_STW_SEL_PRE_L2_SIZE,
+	EU_STW_SEL_LAST    = EU_STW_SEL_PRE_L2_SIZE,
 };
 
 #define STW_NEXT_LAYER(layer)   ((layer) - 1)
@@ -833,13 +833,11 @@ static inline bool mod_is_dpu_rcp_valid(u32 reg)
 static inline bool __mod_is_layer_valid(enum mod_eu_stw_mux_sel off,
 					const char *caller)
 {
-	if (likely((off >= EU_STW_SEL_L3_OFF5 && off <= EU_STW_SEL_L3_OFF0) ||
-		   off == EU_STW_SEL_L2_OFF))
+	if (likely((off >= EU_STW_SEL_L3_OFF5 && off <= EU_STW_SEL_L3_OFF0)))
 		return true;
 
-	pr_err("Invalid layer off %d, valid values: %d..%d or %d, caller %s\n",
-	       off, EU_STW_SEL_L3_OFF5, EU_STW_SEL_L3_OFF0,
-	       EU_STW_SEL_L2_OFF, caller);
+	pr_err("Invalid layer off %d, valid values: %d..%d, caller %s\n", off,
+	       EU_STW_SEL_L3_OFF5, EU_STW_SEL_L3_OFF0, caller);
 	return false;
 }
 
